src/main/java/com/redhat/devtools/lsp4ij/features/completion/SnippetTemplateLoader.java

```diff
@@ -32,43 +37,129 @@
 public class SnippetTemplateLoader extends AbstractLspSnippetHandler {
 
     private final Template template;
+    private final boolean useTemplateForInvocationOnlySnippet;
 
-    private final List<String> existingVariables;
+    private final List<SnippetTemplateSegment> templateSegments = new LinkedList<>();
+    private final List<String> existingVariables = new LinkedList<>();
 
     /**
      * Load the given Intellij template from a LSP snippet content by using the given variable resolver.
      *
-     * @param template         the Intellij template to load.
-     * @param variableResolver the variable resolver (ex : resolve value of TM_SELECTED_TEXT when snippet declares ${TM_SELECTED_TEXT})
-     * @param indentOptions the LSP indent options to replace '\t' and '\n' characters according the code style settings of the language.
+     * @param template                            the Intellij template to load.
+     * @param variableResolver                    the variable resolver (ex : resolve value of TM_SELECTED_TEXT when snippet declares ${TM_SELECTED_TEXT})
+     * @param indentOptions                       the LSP indent options to replace '\t' and '\n' characters according the code style settings of the language.
+     * @param useTemplateForInvocationOnlySnippet whether or not a template should be used for an invocation-only snippet
      */
-    public SnippetTemplateLoader(Template template, Function<String, String> variableResolver, LspSnippetIndentOptions indentOptions) {
+    public SnippetTemplateLoader(@NotNull Template template,
+                                 @Nullable Function<String, String> variableResolver,
+                                 @Nullable LspSnippetIndentOptions indentOptions,
+                                 boolean useTemplateForInvocationOnlySnippet) {
         super(variableResolver, indentOptions);
         this.template = template;
-        this.existingVariables = new ArrayList<>();
+        this.useTemplateForInvocationOnlySnippet = useTemplateForInvocationOnlySnippet;
     }
 
     @Override
     public void startSnippet() {
 
     }
 
+    @SuppressWarnings("DataFlowIssue")
     @Override
     public void endSnippet() {
+        List<SnippetTemplateSegment> effectiveTemplateSegments = templateSegments;
+
+        // If we should not use a template for an invocation-only snippet, see whether this fits the pattern
+        if (!useTemplateForInvocationOnlySnippet) {
+            boolean hasInvocation = false;
+            // While it's unlikely to happen in a typical invocation arg list scenario, this helps ensure we're only
+            // reacting to the contents of balanced parentheses
+            int nestedInvocationCount = 0;

```

Comment:
This is an additional safeguard against the unlikely possibility of nested parens in segments. We'll only come out of this logic deciding to use the simplified text segments if everything occurred within balanced invocation parens.